core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -1418,4 +1421,36 @@ object FiberRuntime {
     }
 
   private val notBlockingOn: () => FiberId = () => FiberId.None
+
+  private final class Inbox {
+    private[this] val queue    = new java.util.concurrent.ConcurrentLinkedQueue[FiberMessage]()
+    private[this] var _isEmpty = true
```

Comment:
Should this be made volatile? The only scenario that kind of worries me is the following:

1. Thread 1 (runLoop thread) empties the queue (this implies that a message was added right in the previous iteration as well)
2. Thread 1: calls `queue.isEmpty` and sets `_isEmpty` to `true`
3. Thread 2 (external) adds message to queue immediately after
4. Thread 2: sets `_isEmpty` to false but somehow (2) is executed afterwards

The chances of this exact sequence of events happening are astronomically small to begin with, but is it something we need to cater for? 🤔

Reply:
@jdegoes I see your point! I'll revert this change in the PR since it's now tracked by https://github.com/zio/zio/issues/8807 and give it a go at tackling it separately